lib/rouge/lexers/bbcbasic.rb

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class BBCBASIC < RegexLexer
+      title "BBCBASIC"
+      desc "BBC BASIC syntax"
+      tag 'bbcbasic'
+      filenames '*,fd1'
+
+      def self.punctuation
+        @punctuation ||= %w(
+          [:,;'~] SPC TAB
+        )
+      end
+
+      def self.function
+        @function ||= %w(
+          ABS ACS ADVAL ASC ASN ATN BEATS BEAT BGET# CHR\$ COS COUNT DEG EOF#
+          ERL ERR ERROR EVAL EXP EXT# GET GET\$ GET\$# HIMEM INKEY INKEY\$
+          INSTR INT LEFT\$ LEN LN LOG LOMEM MID\$ OPENIN OPENOUT OPENUP PAGE
+          POS PTR# RAD REPORT\$ RIGHT\$ RND SGN SIN SQR STR\$ STRING\$ SUM
+          SUMLEN TAN TEMPO TIME TIME\$ TOP USR VAL VPOS
+        )
+      end
+
+      def self.control
+        @control ||= %w(
+          CASE CHAIN ELSE ENDCASE ENDIF ENDPROC ENDWHILE END FN FOR GOSUB GOTO
+          IF INSTALL LIBRARY NEXT OF OTHERWISE OVERLAY PROC REPEAT RETURN STEP
+          STOP THEN TO UNTIL WHEN WHILE
+        )
+      end
+
+      def self.statement
+        @statement ||= %w(
+          BEATS BPUT# CALL CLEAR CLG CLOSE# CLS COLOR COLOUR DATA DIM ENVELOPE
+          GCOL LET MODE OFF ON ORIGIN OSCI PLOT PRINT PRINT# QUIT READ REPORT
+          SOUND STEREO SWAP SYS TINT VDU VOICE VOICES WAIT WIDTH
+        )
+      end
+
+      def self.operator
+        @operator ||= %w(
+          << <= <> < >= >>> >> > [-!\$()*+\/=?^|] AND DIV EOR MOD NOT OR
+        )
+      end
+
+      def self.constant
+        @constant ||= %w(
+          FALSE TRUE
+        )
+      end
+
+      state :root do
+        rule %r/[ \n]+/, Text
+        rule %r/[\[]/, Keyword, :assembly1
+        rule %r/\*.*/, Generic::Prompt # CLI command
```

Comment:
What does this represent? `Generic::Prompt` seems like a strange token to be using.

Reply:
It's a way of inlining a command-line command, and by a not-accidental coincidence, `*` is the default command line prompt on RISC OS. Now that I'm aware that we can colour different parts of the matched string according to which capture group they're part of, I'm tempted to make only the `*` `Generic::Prompt` and have the rest be `Text`.

Reply:
Hmmm, that now makes sense but I'm not sure it's really the right token to be using here. My reading of [the Pygments documentation](http://pygments.org/docs/tokens/) is that the token is really more for situations where you're highlighting text that _is_ a command prompt. I think the more appropriate token here is `Operator`.

Reply:
(Although I don't have a strong opinion that `Operator` is correct, more that I don't think `Generic::Prompt` is correct.)

Reply:
I reckoned that making it `Keyword` was probably better, so as to distinguish it from the mutiplication operator. However, having made that change and tested it, I seem to have opened a can of worms, because `*` was matching as a command introducer far more often than it should do. You should see I've had a go at rectifying that by adding an extra state, to restrict the locations at which `*` can be a command introducer. It's still not perfect though: ideally, when in `:root` state, if it hits either of the `mixin` rules, I'd want to transition to `:no_further_imperatives`. I'm not sure how to do that, any ideas?

Reply:
Looks like, per the below, this is working now.